ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVTest.scala

```diff
@@ -199,59 +172,63 @@ object KVTest {
       submitter: Party,
       submissionSeed: crypto.Hash,
       cmds: Command*,
-  ): KVTest[(SubmittedTransaction, Tx.Metadata)] =
-    runCommand(submitter, submissionSeed, defaultAdditionalContractDataTy, cmds: _*)
+  ): KVTest[(SubmittedTransaction, Transaction.Metadata)] =
+    runCommand(submitter, submissionSeed, DefaultAdditionalContractDataTy, cmds: _*)
 
   def submitTransaction(
       submitter: Party,
-      transaction: (SubmittedTransaction, Tx.Metadata),
+      transaction: (SubmittedTransaction, Transaction.Metadata),
       submissionSeed: crypto.Hash,
       letDelta: Duration = Duration.ZERO,
       commandId: CommandId = randomLedgerString,
       deduplicationTime: Duration = Duration.ofDays(1)): KVTest[(DamlLogEntryId, DamlLogEntry)] =
     for {
       testState <- get[KVTestState]
-      submInfo = SubmitterInfo(
-        submitter = submitter,
-        applicationId = Ref.LedgerString.assertFromString("test"),
-        commandId = commandId,
-        deduplicateUntil =
-          testState.recordTime.addMicros(deduplicationTime.toNanos / 1000).toInstant,
-      )
+      submInfo = createSubmitterInfo(submitter, commandId, deduplicationTime, testState)
       (tx, txMetaData) = transaction
-      subm = keyValueSubmission.transactionToSubmission(
-        submitterInfo = submInfo,
-        meta = TransactionMeta(
-          ledgerEffectiveTime = testState.recordTime.addMicros(letDelta.toNanos / 1000),
-          workflowId = None,
-          submissionTime = txMetaData.submissionTime,
-          submissionSeed = submissionSeed,
-          optUsedPackages = Some(txMetaData.usedPackages),
-          optNodeSeeds = None,
-          optByKeyNodes = None,
-        ),
-        tx = tx
-      )
+      subm = transactionToSubmission(submissionSeed, letDelta, testState, submInfo, tx, txMetaData)
       result <- submit(subm)
     } yield result
 
-  val minMRTDelta: Duration = Duration.ofSeconds(1)
+  def preExecuteTransaction(
+      submitter: Party,
+      transaction: (SubmittedTransaction, Transaction.Metadata),
+      submissionSeed: crypto.Hash,
+      letDelta: Duration = Duration.ZERO,
+      commandId: CommandId = randomLedgerString,
+      deduplicationTime: Duration = Duration.ofDays(1))
+    : KVTest[(DamlLogEntryId, PreExecutionResult)] =
+    for {
+      testState <- get[KVTestState]
+      submInfo = createSubmitterInfo(submitter, commandId, deduplicationTime, testState)
+      (tx, txMetaData) = transaction
+      subm = transactionToSubmission(submissionSeed, letDelta, testState, submInfo, tx, txMetaData)
+      result <- preExecute(subm)
+    } yield result
 
   def submitConfig(
       configModify: Configuration => Configuration,
       submissionId: SubmissionId = randomLedgerString,
-      mrtDelta: Duration = minMRTDelta
+      mrtDelta: Duration = MinMaxRecordTimeDelta
```

Comment:
For consistency you might want to remove all occurrences of `mrtDelta` parameter to `minMaxRecordTimeDelta`.

Reply:
Done.